envoy/http/filter.h

```diff
@@ -812,6 +812,11 @@ class StreamDecoderFilterCallbacks : public virtual StreamFilterCallbacks,
    */
   virtual absl::optional<Upstream::LoadBalancerContext::OverrideHost>
   upstreamOverrideHost() const PURE;
+
+  /**
+   * @return true if the filter should shed load based on the system pressure, typically memory.
+   */
+  virtual bool shouldLoadShed() const { return false; };
```

Comment:
Is it too big of a refactor if you make it PURE?

Reply:
Nope, good point, done.